internal/pkg/kubernetes/kubernetes.go

```diff
@@ -0,0 +1,113 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package kubernetes
+
+import (
+	"log"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	policy "k8s.io/api/policy/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	"k8s.io/client-go/kubernetes"
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Helper represents a set of helper methods for interacting with the
+// Kubernetes API.
+type Helper struct {
+	client *kubernetes.Clientset
+}
+
+// NewHelper initializes and returns a Helper.
+func NewHelper() (helper *Helper, err error) {
+	kubeconfig := "/etc/kubernetes/kubelet.conf"
+
+	var config *restclient.Config
+	config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	if err != nil {
+		return nil, err
+	}
+
+	var clientset *kubernetes.Clientset
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Helper{clientset}, nil
+}
+
+// CordonAndDrain cordons and drains a node in one call.
+func (h *Helper) CordonAndDrain(node string) (err error) {
+	if err = h.Cordon(node); err != nil {
+		return err
+	}
+	return h.Drain(node)
+}
+
+// Drain evicts all pods on a given node.
+func (h *Helper) Drain(node string) error {
+	opts := meta.ListOptions{
+		FieldSelector: fields.SelectorFromSet(fields.Set{"spec.nodeName": node}).String(),
+	}
+	pods, err := h.client.CoreV1().Pods(meta.NamespaceAll).List(opts)
+	if err != nil {
+		return errors.Wrapf(err, "cannot get pods for node %s", node)
+	}
+
+	var wg sync.WaitGroup
+	wg.Add(len(pods.Items))
+
+	// Evict each pod.
+
+	gracePeriod := int64(60)
+	for _, pod := range pods.Items {
+		func(p corev1.Pod) {
```

Comment:
should we kick this off in a goroutine ( do a mass exodus at once ) ?

Reply:
whoops... that is what I meant to do :D